Makefile

```diff
@@ -0,0 +1,24 @@
+CXX ?= g++
+
+COMPILE_FLAGS += -W -Wall -Wextra -Werror -O0 -Wno-unused-parameter -fsanitize=address,undefined -g
```

Comment:
Maybe use `ifdef $DEBUG)` to add those debugging flags, and you can use `make DEBUG=1` to compile with debug symbols and sanitizers.